### ISSUE TITLE
16_tied.t: free to wrong pool

### ISSUE THREAD

**Issue by rurban** [rurban/Cpanel-JSON-XS#168]
The new code has a free problem. valgrind it
```
1..4
ok 1
==2821424== Invalid write of size 8
==2821424==    at 0x127CA927: encode_hv (XS.xs:1343)
==2821424==    by 0x127D0503: encode_sv (XS.xs:1878)
==2821424==    by 0x127D7D6A: encode_json (XS.xs:2632)
==2821424==    by 0x127E6717: XS_Cpanel__JSON__XS_encode (XS.xs:4767)
==2821424==    by 0x592AE0: Perl_pp_enterxssub (pp_hot.c:5700)
==2821424==    by 0x7D6C1C: Perl_runops_debug (dump.c:3124)
==2821424==    by 0x42837A: S_run_body (perl.c:2860)
==2821424==    by 0x42790E: perl_run (perl.c:2767)
==2821424==    by 0x421D1C: main (perlmain.c:137)
==2821424==  Address 0x128437f8 is 0 bytes after a block of size 8 alloc'd
==2821424==    at 0x483BB1A: calloc (vg_replace_malloc.c:760)
==2821424==    by 0x127CA8C4: encode_hv (XS.xs:1341)
==2821424==    by 0x127D0503: encode_sv (XS.xs:1878)
==2821424==    by 0x127D7D6A: encode_json (XS.xs:2632)
==2821424==    by 0x127E6717: XS_Cpanel__JSON__XS_encode (XS.xs:4767)
==2821424==    by 0x592AE0: Perl_pp_enterxssub (pp_hot.c:5700)
==2821424==    by 0x7D6C1C: Perl_runops_debug (dump.c:3124)
==2821424==    by 0x42837A: S_run_body (perl.c:2860)
==2821424==    by 0x42790E: perl_run (perl.c:2767)
==2821424==    by 0x421D1C: main (perlmain.c:137)
==2821424== 
==2821424== Invalid read of size 8
==2821424==    at 0x127C8755: he_cmp_tied (XS.xs:1232)
==2821424==    by 0x4A4DE74: msort_with_tmp.part.0 (in /usr/lib64/libc-2.30.so)
==2821424==    by 0x4A4DDE4: msort_with_tmp.part.0 (in /usr/lib64/libc-2.30.so)
==2821424==    by 0x4A4E345: qsort_r (in /usr/lib64/libc-2.30.so)
==2821424==    by 0x127CAAFB: encode_hv (XS.xs:1373)
==2821424==    by 0x127D0503: encode_sv (XS.xs:1878)
==2821424==    by 0x127D7D6A: encode_json (XS.xs:2632)
==2821424==    by 0x127E6717: XS_Cpanel__JSON__XS_encode (XS.xs:4767)
==2821424==    by 0x592AE0: Perl_pp_enterxssub (pp_hot.c:5700)
==2821424==    by 0x7D6C1C: Perl_runops_debug (dump.c:3124)
==2821424==    by 0x42837A: S_run_body (perl.c:2860)
==2821424==    by 0x42790E: perl_run (perl.c:2767)
==2821424==  Address 0x128437f8 is 0 bytes after a block of size 8 alloc'd
==2821424==    at 0x483BB1A: calloc (vg_replace_malloc.c:760)
==2821424==    by 0x127CA8C4: encode_hv (XS.xs:1341)
==2821424==    by 0x127D0503: encode_sv (XS.xs:1878)
==2821424==    by 0x127D7D6A: encode_json (XS.xs:2632)
==2821424==    by 0x127E6717: XS_Cpanel__JSON__XS_encode (XS.xs:4767)
==2821424==    by 0x592AE0: Perl_pp_enterxssub (pp_hot.c:5700)
==2821424==    by 0x7D6C1C: Perl_runops_debug (dump.c:3124)
==2821424==    by 0x42837A: S_run_body (perl.c:2860)
==2821424==    by 0x42790E: perl_run (perl.c:2767)
==2821424==    by 0x421D1C: main (perlmain.c:137)
```

**Comment by rurban** [comment]
The allocation of the temp. HEK was not big enough. Forgot about that.

**Comment by rurban** [comment]
Fixed with fbb1a24ca0fada6a97143d4d01676e40f9e46dd7
in 4.21 btw
